proptest/src/test_runner/failure_persistence/mod.rs

```diff
@@ -142,15 +152,36 @@ impl Clone for Box<dyn FailurePersistence> {
     }
 }
 
+pub(crate) fn to_base16(dst: &mut String, src: &[u8]) {
+    for byte in src {
+        dst.push_str(&format!("{:02x}", byte));
+    }
+}
+
+pub(crate) fn from_base16(dst: &mut [u8], src: &str) -> Option<()> {
```

Comment:
this function looks a bit awkward because the return value is easy to ignore given `dst`, what if we returned a result that contained the dst value instead?

maybe something like:
```rust
pub(crate) fn from_base16<const N: usize>(
    src: &str,
) -> Result<[u8; N], &'static str> {
    if src.len() != N * 2 {
        return Err("Input length does not match expected buffer size.");
    }

    let mut buffer = [0u8; N];

    for (i, chunk) in src.as_bytes().chunks(2).enumerate() {
        let hex_str = std::str::from_utf8(chunk)
            .map_err(|_| "Input contains invalid UTF-8.")?;
        buffer[i] = u8::from_str_radix(hex_str, 16)
            .map_err(|_| "Failed to parse hex chunk as hexadecimal.")?;
    }

    Ok(buffer)
}
```